StripePaymentSheet/StripePaymentSheet/Source/Helpers/IntentStatusPoller.swift

```diff
@@ -30,72 +35,69 @@ class IntentStatusPoller {
         didSet {
             // Start polling if we weren't already polling
             if !oldValue && isPolling {
-                forcePoll()
+                fetchStatus()
             } else if !isPolling {
                 nextPollWorkItem?.cancel()
             }
         }
     }
 
-    init(apiClient: STPAPIClient, clientSecret: String, maxRetries: Int) {
-        self.apiClient = apiClient
+    init(retryInterval: TimeInterval, intentRetriever: PaymentIntentRetrievable, clientSecret: String) {
+        self.retryInterval = retryInterval
+        self.intentRetriever = intentRetriever
         self.clientSecret = clientSecret
-        self.maxRetries = maxRetries
     }
 
     // MARK: Public APIs
 
+    /// Begins continuously polling at a frequency defined by `retryInterval`. The delegate will be
+    /// notified of any status changes to the intent during this poll period.
     public func beginPolling() {
         isPolling = true
     }
 
+    /// Suspends the ongoing polling process. After suspending, the delegate will not receive any status
+    /// updates on the intent until polling is resumed using `beginPolling()`.
     public func suspendPolling() {
         isPolling = false
     }
 
-    public func forcePoll() {
-        fetchStatus(forcePoll: true)
+    /// Triggers a single poll operation to fetch the intent status right away, regardless of the
+    /// retryInterval. Whether or not continuous polling in ongoing, `pollOnce()` immediately updates the
+    /// status and informs the delegate if a change in status takes place.
```

Comment:
How does the user of this class know when polling is completed? 

Like, when the timer runs out, how does PollingViewController know that the last network request completed and it should dismiss itself?

Reply:
The user of this class doesn't know when polling is completed since that is up to the user/timer. The idea here is the IntentStatusPoller just polls until you ask it to stop. So you keep polling until you have received the status you are looking for or want to cancel, in which case you call `suspendPolling`.

Reply:
Well let's say I'm the PollingVC and I want to poll for 60 seconds and then dismiss and return to PaymentSheet.  

After 60 seconds, how do I know when the final request has completed so I can dismiss myself?  

In fact, looking at the code more, it doesn't appear to dismiss at all when the timer expires. It only dismisses when the PI status changes or you cancel out?

Reply:
The original thought was that if the timer expires on the client, then on the server the intent would move out of `requires_action` back into `requires_payment_method` which would notify the delegate, in which case we show the error/timed out UI in `didUpdate(paymentIntent: STPPaymentIntent)`. But it probably makes sense to show that UI if the final poll doesn't come back as succeeded as a safe guard. I'm going to add a completion handler to `pollOnce` and add that logic.

https://www.figma.com/file/C6FkqGHZuSzR6S0KoseV0T/UPI-Payment?type=design&node-id=24-2762&mode=design&t=gOIJDDZFqkZ3Z8qX-0

Reply:
I think the change you made fixes the problem I'm describing.